vizro-core/docs/pages/user-guides/data.md

```diff
@@ -333,14 +336,77 @@ Parametrized data loading is compatible with [caching](#configure-cache). The ca
 
 You cannot pass [nested parameters](parameters.md#nested-parameters) to dynamic data. You can only target the top-level arguments of the data loading function, not the nested keys in a dictionary.
 
-### Filter update limitation
+### Filters
+
+By default, when a [filter](filters.md) depends on dynamic data, the available selector values update on page refresh to reflect the latest dynamic data. The mechanism for this, including caching, works exactly like other non-control components such as `vm.Graph`. However, unlike such components, a filter can depend on multiple data sources.
+
+Vizro looks at the data source for each component in the filter's `targets`. If at least one of these is a dynamic data source then the filter behaves dynamically and the filter's selector updates on page refresh. Remember that when `targets` is not explicitly specified, a filter applies to all the components on a page that use a DataFrame including `column`.
```

Comment:
Just to clarify: Suppose I have a page with charts connected to two different data sources, both of which have a year column. I add a Slider with `year` as the column. Let's say data source A is dynamic, while data source B is static.

On the initial load, the Slider will show all years from both data A and data B. Now, if data source A updates and has fewer years, then on page refresh, the Slider will display the **updated years from data A** and **all the years from data B.** Correct? It is **not the case** that now only the updated years from data A are shown, but still the union of the two.

Something like this:

```
import vizro.models as vm
import vizro.plotly.express as px
from vizro import Vizro

gapminder = px.data.gapminder().query("country == 'Nigeria' and year > 1990")
gapminder_two = px.data.gapminder().query("country == 'Nigeria' and year <= 1990")

bar = px.bar(gapminder,  x="year", y="lifeExp")
bar_two = px.bar(gapminder_two,  x="year", y="lifeExp", color_discrete_sequence=["violet"])

page = vm.Page(title="My page",
               components=[vm.Graph(figure=bar), vm.Graph(figure=bar_two)],
               controls=[vm.Filter(column="year")])

dashboard = vm.Dashboard(pages=[page])

if __name__ == "__main__":
    Vizro().build(dashboard).run()
```

Reply:
In other words, it always shows the **intersection** of the the values of (A,B), but it shows the previously selected options right?

Reply:
Done!